### DISCUSSION TITLE
How to capture arguments from command line argument in workflow

### DISCUSSION THREAD

**Discussion by yamaura** [nextflow-io/nextflow#3515] [Q&A] [answered]
Hi!

I want to capture value from arguments not options.
This helps me to launch workflow with following commands:

```
$ nextflow test.nf --test arg1 read{1,2}.fq
```

I can catch `arg1` value with `params.test`.
But how to capture the values `raed1.fq` and `read2.fq`?

Thanks

**Comment by mribeirodantas** [comment, marked as answer]
If you want to provide several values to a parameter in Nextflow, you can either wrap them with quotes or concatenate them with a comma. See the examples below:

`example.nf`
```
println params.example
```
You can run this Nextflow script in the two ways below to get all values:
```
nextflow run example.nf --example test1,test2,test3
# OR
nextflow run example.nf --example "test1 test2 test3"
```

Output:
```
➜  ~ nextflow run example.nf --example "test test2 test3"
N E X T F L O W  ~  version 22.10.4
Launching `ex.nf` [scruffy_payne] DSL2 - revision: a982a55670
test test2 test3
```
Or
```
➜  ~ nextflow run example.nf --example test,test2,test3
N E X T F L O W  ~  version 22.10.4
Launching `ex.nf` [tender_heyrovsky] DSL2 - revision: a982a55670
test,test2,test3
```

You should also check the `-params-file` argument of the Nextflow CLI. You can provide a JSON/YAML file to it and access the fields within your Nextflow script. For example, let's replace our `example.nf` file with the following content:
```
println params.example1
println params.example2
```
And then create a file named `my_params.json` with the following content:
```
{
  "example1": "Test 1",
  "example2": "Test 2"
}
```
Now, when we run `nextflow run example.nf -params-file my_params.json` I will get:
```
➜  ~ nextflow run ex.nf -params-file my_params.json
N E X T F L O W  ~  version 22.10.4
Launching `ex.nf` [boring_bhaskara] DSL2 - revision: b4dc14b49e
Test 1
Test 2
```

**Reply by pditommaso** [reply]
or just use the `args` implicit variable using the first command line 

For example having `test.nf` as: 

```
println "test: $params.test"
println "args: $args"
```

```
» nextflow run test.nf --test arg1 read{1,2}.fq
N E X T F L O W  ~  version 22.12.0-edge
Launching `test.nf` [deadly_jones] DSL2 - revision: 75203f325f
test: arg1
args: [read1.fq, read2.fq]
```

**Reply by mribeirodantas** [reply]
That's a new one for me! That's nice! Do you have a good use case in mind? Or know of anyone using it to do something cool?

Also, can I separate it between parameters?
```
➜  ~ nextflow run ex.nf --example1 test1 blah blah --example2 test2 bleh bleh   
N E X T F L O W  ~  version 22.10.4
Launching `ex.nf` [sharp_swanson] DSL2 - revision: 439718d4f5
test1
test2
[blah, blah, bleh, bleh]
```

I would like to get a list with `blah, blah` and another one with `bleh, bleh`.

**Reply by pditommaso** [reply]
`args` is just an array holding all unnamed cli parameters. therefore you can `args[0]` to access therfirst, `args[1]` the second, .. etc

**Reply by mribeirodantas** [reply]
Oh, OK. Now I get how it works. Thanks 😄

**Comment by yamaura** [comment]
Thank you, It is what I wanted to know!
